segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeperstorage/StorageLogManager.java

```diff
@@ -0,0 +1,813 @@
+                                            currentOffset.set((long) 0);
+                                        }
+                                    } catch (IOException e) {
+                                        future.completeExceptionally(new CompletionException(e));
+                                        return;
+                                    }
+                                    if (entry.getEntryId() == ledgerHandle.getLastAddConfirmed()) {
+                                        future.complete(null);
+                                        readingDone.set(true);
+                                    }
+                                }
+                                firstEntryId.set(firstEntryId.get() + entriesInOneRound + 1);
+                                future.complete(null);
+                            }, future);
+                    return future;
+                },
+                executor)
+                            .thenApply(v -> length - lengthRemaining.get());
+    }
+
+    private CompletableFuture<StorageLog> fenceLedgersAndCreateOneAtEnd(String streamSegmentName, StorageLog ledger) {
+        return fenceAllTheLedgers(streamSegmentName, ledger)
+                .thenCompose(storageLog -> {
+                    log.info("Made all the ledgers readonly. Adding a new ledger at {} for {}",
+                            storageLog.getLength(), streamSegmentName);
+                    return createLedgerAt(streamSegmentName, (int) storageLog.getLength())
+                            .thenApply(ledgerData -> {
+                                storageLog.addToList((int) storageLog.getLength(), ledgerData);
+                                return storageLog;
+                            });
+                });
+    }
+
+    private CompletableFuture<StorageLog> fenceAllTheLedgers(String streamSegmentName, StorageLog ledger) {
+        return zkClient.getChildren().forPath(getZkPath(streamSegmentName))
+                       .toCompletableFuture()
+                       .exceptionally(exc -> {
+                           translateZKException(streamSegmentName, exc);
+                           return null;
+                       })
+                       .thenCompose(
+                               children -> {
+                                   if (children.size() == 0) {
+                                       return createLedgerAt(streamSegmentName, 0).thenApply(ledgerData -> {
+                                           ledger.addToList(0, ledgerData);
+                                           return ledger;
+                                       });
+                                   } else {
+                                       CompletableFuture<LedgerData>[] futures = children.stream()
+                                                                                         .map(child -> {
```

Comment:
Can we do a better indentation of this chunk of code? It overflowing way too much to the right.

Reply:
Fixed.